[revdrag] iP

revdrag opened the pull request:
# Ren Chatbot

> "An investment in knowledge pays the best interest." - Benjamin Franklin ([Source](https://www.brainyquote.com/quotes/benjamin_franklin_141119))

Ren frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is, 
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double-click it
3. add your tasks
4. let it manage your tasks for you! :star_struck:

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon!)
***
If you are a Java Programmer, you can use it practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

revdrag committed:
Implemented Level-0

revdrag committed:
Implemented Level-1

revdrag committed:
Implemented Level-2

revdrag committed:
Fixed a bug with level-2

revdrag committed:
Implemented Level-3 and A-Classes

revdrag committed:
Implemented Level-4

revdrag committed:
Implemented A-TextUiTesting and added some extra lines to Duke

revdrag committed:
Implemented Level-5 and A-Exceptions

revdrag committed:
Implemented Level-6

revdrag committed:
Add reading file logic

revdrag committed:
Implement branch-Level-7

revdrag committed:
This reverts commit f380b9cd356ab1b98c6c0c147e3f3704a1e69ba1.

revdrag committed:
Update Level-7

revdrag committed:
Merge branch 'branch-Level-7'

revdrag committed:
Fix Level-7

revdrag committed:
Implement Level-8

revdrag committed:
Merge branch 'branch-Level-8'

revdrag committed:
Implement A-MoreOOP

revdrag committed:
Merge branch 'branch-A-MoreOOP'

revdrag committed:
Implement A-Packages

revdrag committed:
Merge branch 'branch-A-Packages'

revdrag committed:
Merge branch 'add-gradle-support' of https://github.com/revdrag/ip into add-gradle-support

revdrag committed:
Merge branch 'add-gradle-support'

revdrag committed:
Merge branch 'add-gradle-support' of https://github.com/revdrag/ip into add-gradle-support

revdrag committed:
Merge branch 'add-gradle-support'

revdrag committed:
Add Gradle Support

revdrag committed:
Merge branch 'branch-A-Gradle'

revdrag committed:
Merge branch 'master' of https://github.com/revdrag/ip

revdrag committed:
Add 2 JUnit Tests for TaskList and Storage

revdrag committed:
Merge branch 'branch-A-JUnit'

revdrag committed:
Add JavaDocs/ header comments to most of the public methods.

revdrag committed:
Fix some portions of the code that did not follow CodingStandard

revdrag committed:
Implement Level-9, Add find function to Duke

revdrag committed:
Merge branch 'branch-A-CodingStandard'

revdrag committed:
Merge branch 'branch-A-JavaDoc'

revdrag committed:
Merge branch 'branch-Level-9'

kanna-1 commented on src/main/java/duke/DateHelper.java:
```diff
@@ -0,0 +1,38 @@
+     * Returns LocalDateTime from a String of format dd-MM-yyyy HHmm
+     *
+     * @param datetime String input
+     * @return LocalDateTime
```
Missing punctuations behind each parameter description and method description.

kanna-1 commented on src/main/java/duke/DateHelper.java:
```diff
@@ -0,0 +1,38 @@
+     * Returns a different datetime format as a String from inputted format.
+     *
+     * @param parsedTime Parsed LocalDateTime from user input
+     * @return Different String datetime format
```
Missing punctuations behind each parameter description.

kanna-1 commented on src/main/java/duke/DateHelper.java:
```diff
@@ -0,0 +1,38 @@
+     * Returns datetime as a String in the format dd-MM-yyyy HHmm to be used to save into data file.
+     *
+     * @param parsedTime Parsed LocalDateTime from user input
+     * @return String format in dd-MM-yyyy HHmm
```
... I noticed the same punctuation issue in several other places too.

kanna-1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+
+    public Parser() {
+
+    }
```
Is this necessary since in Java if you don't define any constructors in your class, the Java compiler automatically provides a default constructor for you.

revdrag replied:
Thanks for noticing that!

kanna-1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+     */
+    public void parse(String command, TaskList tasks) {
+        try {
+            String[] commandArr = command.split(" ",2);
```
The whitespace format is incorrect.

kanna-1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+                    Ui.bye();
+                    break;
+                default:
+                    throw new DukeException(DukeException.UNKNOWN);
```
The switch statement format is incorrect.

It should have the following form: 
```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}

kanna-1 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasklist;
```
Should `tasklist` be in camelCase?

kanna-1 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,56 @@
+     * @param target The String to be found.
+     * @return True or False depending on whether the target is found.
+     */
+    public boolean find(String target) {
```
Boolean methods should be named to sound like booleans. For example, `isFound`.

kanna-1 commented on src/main/java/duke/DateHelper.java:
```diff
@@ -0,0 +1,38 @@
+     * @param datetime String input
+     * @return LocalDateTime
+     */
+    public static LocalDateTime parse(String datetime) {
```
Should `datetime` be in camelCase?

kanna-1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+    private Storage storage;
+    final static String LINE = "────────────────────────────────────────────────────";
+
+    public Duke(String filepath) {
```
Should `filepath` be in camelCase?

kanna-1 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+
+public class Storage {
+    private File dataFile;
+    private String filepath;
+
+    public Storage(String filepath) {
+        this.filepath = filepath;
+        this.dataFile = new File(filepath);
```
Should `filepath` be in camelCase?

revdrag committed:
Added CheckStyle into Gradle

revdrag committed:
Merge branch 'branch-A-CheckStyle'

kanna-1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+    /**
+     * {@inheritDoc}
+     */
```
Good use of {@inheritDoc} to efficiently inherit documentation.

kanna-1 reviewed (commented):
Overall good work! Only a few minor coding standard violations I can find! 😄

revdrag committed:
Implemented GUI for Dukegit add . (with bugs which will be fixed in the future)

revdrag committed:
Merge branch 'branch-Level-10'

revdrag committed:
Refactor files into duke.management package.

revdrag committed:
Fix command "bye" bug

Bye command have these bugs:
* Did not close Duke after user inputs bye
* Did not save the updated tasklist into data file

Fixing the bug would allow user's to properly update
their tasklist as well as exit the program.

Let's change the logic in Duke's class to accomodate
a writeToFile() method which will be called in
MainWindow when user's response is equals to the "bye"
message.

revdrag committed:
Fixed closing of duke logic

revdrag committed:
Merge branch 'master' of https://github.com/revdrag/ip

revdrag committed:
Add Java Assertions to Duke

Use Java assert method to help with development of Duke.

The Assert keyword allows me to quickly verify certain assumptions made
during the running of Duke.

Let's add assert in these areas,
* In Task.java, markAsDone() and markAsUndone() has underlying
  assumptions that the boolean value of isDone is false and true
  respectively.
* In Duke.java, getResponse(String) assumes there is always an input by
  the user which might not be true.

Using assert will help to quickly verify such assumptions are checked.

revdrag committed:
Update codebase to improve code quality

Boolean methods should sound like a Boolean to improve readability.
Numerous changes to JavaDoc comments to improve documentation.
Standardise import statements to follow the format:
1. package name
2. importing of Java libraries
3. importing of classes from other packages

Let's go through all the classes and make the changes.
* refactor all import statements to follow the above format
* refactor all JavaDoc comments to be updated to changed methods as well
  as methods and classes with no JavaDoc
* in Task, update find(String target) to isFound(String target) to make
  it sound like a boolean

revdrag committed:
Merge pull request #1 from revdrag/branch-A-Assertions

Add Java Assertions to Duke

revdrag committed:
Add whitespace before { in Storage class

revdrag committed:
Fix merge conflicts

revdrag committed:
Merge pull request #2 from revdrag/branch-A-CodeQuality

Update codebase to improve code quality

revdrag committed:
Add support to add notes using note command

revdrag committed:
Merge branch 'branch-D-Notes'

revdrag committed:
Update styling for DialogBox

revdrag committed:
Fix code quality, Add personality and more error handling

revdrag committed:
Update and add new JUnit Tests

revdrag committed:
Improve GUI

revdrag committed:
Partially update README

revdrag committed:
Update README.md

revdrag committed:
Update README.md

revdrag committed:
Fix some print issues

revdrag committed:
Add Theme

revdrag committed:
Update README.md

revdrag committed:
Update README.md

revdrag committed:
Change DukeException to reflect better

revdrag committed:
Renamed tasks data file name across the program

revdrag committed:
Update README.md

revdrag committed:
Update README.md

revdrag committed:
Update Save Feature to automatically save

revdrag committed:
Update Gradle
